main/src/library/MutSet.flix

```diff
@@ -337,7 +337,7 @@ namespace MutSet {
     ///
     /// The function `f` must be impure.
     ///
-    pub def foreach(f: a ~> Unit, s: MutSet[a]): Unit & Impure =
+    pub def foreach(f: a -> Unit & ef, s: MutSet[a]): Unit & ef =
```

Comment:
f should have effect `ef` but the whole function should be impure (because it touches mutable memory.)